### ISSUE TITLE
Enable R2R Generics support for Arm platforms

### ISSUE THREAD

**Issue by gkhanna79** [dotnet/coreclr#5181]
We need to enable this for both Arm32 and Arm64. I would suggest starting with the latter.

CC @rahku @kyulee1 @jkotas @JohnChen0 @briansull

**Comment by masonwheeler** [comment]
What is "R2R Generics"?  Googling turns up nothing relevant.

**Comment by gkhanna79** [comment]
R2R is an acronym for Ready2Run that is described at https://github.com/dotnet/coreclr/blob/775003a4c72f0acc37eab84628fcef541533ba4e/Documentation/botr/readytorun-overview.md.

This change is about enabling certain scenarios for Generics to be represented in R2R images.

**Comment by masonwheeler** [comment]
Oh yeah, I vaguely remember reading about this a while back.  Thanks! :)

**Comment by jasonwilliams200OK** [comment]
This abb can probably go in glossary. There is ReadyToRun, could be appended with `(R2R)`? https://github.com/dotnet/coreclr/blob/775003a4/Documentation/project-docs/glossary.md

**Comment by kingces95** [comment]
@gkhanna79 Hello. Forgive me a question a bit off topic, but this is the best spot I found to to ask (please let me know if there a R2R forum that's more appropriate.) Is there a document describing how generics works with R2R? For example, where does the native code live and when is it generated for a generic method in module A which takes as a generic argument a struct defined in module B? For example:

```C#
// Module A
class A { T Identity<T>() { return T; } }

// Module B
struct S { }
class B { void Main() { A<S>(); }
```

Where does `A<S>` live and when does it get generated? Is that the same for the methods of `List<int>`? Are generic instantiations containing value types always JITed at runtime?

**Comment by jkotas** [comment]
> Where does  A<S>  live and when does it get generated?
The code for `A<S>` can only live in A's defining module today.

> Are generic instantiations containing value types always JITed at runtime?
Not always. E.g. if the generic type arguments are in the transitive closure of the generic type defining module, we are able to generate R2R code for it today.

We are planning more post v1 work on making the generic support for R2R better. For example, one of the options being considered is having slower universal code in R2R image that can handle any generic instantiation, and patching it with JITed efficient code for generic instantiations that show up as hot at runtime.

**Comment by danmosemsft** [comment]
@gkhanna79 is this still 2.0? what area label?

**Comment by AndyAyersMS** [comment]
@RussKeldorph @BruceForstall this issue seems stale, can one of you take a look and update?

**Comment by BruceForstall** [comment]
@jkotas @sandreenko AFAIK, R2R is fully enabled for arm/arm64. Is that true? If so, we can close this issue.

**Comment by sandreenko** [comment]
@BruceForstall r2r tests for arm are running in ci, the issues are tracked by #15552.
However, I do not know about "R2R Generics support".

**Comment by jkotas** [comment]
Fixed by #5480 for ARM64 and #13733 for ARM.
